src/client/datascience/kernel-launcher/localKernelFinder.ts

```diff
@@ -189,35 +189,52 @@ export class LocalKernelFinder implements ILocalKernelFinder {
-            // this to start the kernel.
-            const matchingInterpreters = this.findMatchingInterpreters(k, interpreters);
-            if (matchingInterpreters && matchingInterpreters.length) {
-                const result: PythonKernelConnectionMetadata = {
-                    kind: 'startUsingPythonInterpreter',
-                    kernelSpec: k,
-                    interpreter: matchingInterpreters[0],
-                    id: getKernelId(k, matchingInterpreters[0])
-                };
-
-                // If interpreters were found, remove them from the interpreter list we'll eventually
-                // return as interpreter only items
-                filteredInterpreters = filteredInterpreters.filter((i) => !matchingInterpreters.includes(i));
+        const kernelMetadata = await Promise.all(
+            kernelSpecs.map(async (k) => {
+                // Find the interpreter that matches. If we find one, we want to use
+                // this to start the kernel.
+                const matchingInterpreters = this.findMatchingInterpreters(k, interpreters);
+                if (matchingInterpreters && matchingInterpreters.length) {
+                    const result: PythonKernelConnectionMetadata = {
+                        kind: 'startUsingPythonInterpreter',
+                        kernelSpec: k,
+                        interpreter: matchingInterpreters[0],
+                        id: getKernelId(k, matchingInterpreters[0])
+                    };
 
-                // Return our metadata that uses an interpreter to start
-                return result;
-            } else {
-                // No interpreter found. If python, use the active interpreter anyway
-                const result: KernelSpecConnectionMetadata = {
-                    kind: 'startUsingKernelSpec',
-                    kernelSpec: k,
-                    interpreter: k.language === PYTHON_LANGUAGE ? activeInterpreter : undefined,
-                    id: getKernelId(k, activeInterpreter)
-                };
-                return result;
-            }
-        });
+                    // If interpreters were found, remove them from the interpreter list we'll eventually
+                    // return as interpreter only items
+                    filteredInterpreters = filteredInterpreters.filter((i) => !matchingInterpreters.includes(i));
+
+                    // Return our metadata that uses an interpreter to start
+                    return result;
+                } else {
+                    let interpreter = k.language === PYTHON_LANGUAGE ? activeInterpreter : undefined;
+                    // If the interpreter information is stored in kernelspec.json then use that to determine the interpreter.
+                    if (
+                        k.language === PYTHON_LANGUAGE &&
+                        this.extensionChecker.isPythonExtensionInstalled &&
```

Comment:
Even if we had interpreter information in the kernelspec file, we weren't using that. This fixes that.